doc/source/cluster/key-concepts.rst

```diff
@@ -0,0 +1,50 @@
+.. include:: we_are_hiring.rst
+
+.. _cluster-key-concepts:
+
+Key Concepts
```

Comment:
The content you have is great, but the rendered page is overall still a little thin in terms of content. Is this one sentence really all we want users to understand about the autoscaler, for instance?

Reply:
Other topics I think we should touch on: cluster launcher, k8s operator, Job submission, Ray client, Cloud Clusters, Cluster managers.

Reply:
For autoscaler, what else would you add? Details on the algo? I don't have that much in-depth knowledge about it.

I've added all the other ones, except for cloud clusters (wasn't really sure what this means concretely?).

Reply:
We should at least mention the autoscaler's capacity to autoscale.

Specifically these facts or summary thereof: 
If the resource demands of the Ray workload exceed the current capacity of the cluster, the autoscaler will try to add nodes. Conversely, if a node is idle for long enough, the autoscaler will remove it from the cluster. 

One way to convey most of this info is to edit the sentence.
`to match the specification in the cluster config file<cluster-config>` -> `to meet the needs of the Ray workload while matching the specification in the cluster config file<cluster-config>`

Reply:
Added, thanks